core/src/test/scala/org/apache/spark/shuffle/sort/IndexShuffleBlockResolverSuite.scala

```diff
@@ -155,4 +156,19 @@ class IndexShuffleBlockResolverSuite extends SparkFunSuite with BeforeAndAfterEa
       indexIn2.close()
     }
   }
+
+  test("commit shuffle files multiple times") {
+    val shuffleId = 1
+    val mapId = 2
+    val idxName = s"shuffle_${shuffleId}_${mapId}_0.index"
+    testWithIndexShuffleBlockResolver(shuffleId, mapId, idxName, -1)
+  }
+
+  test("commit shuffle files with shuffle generation id multiple times") {
```

Comment:
shall we also test committing shuffle files multiple times with different shuffle generation ids?

Reply:
Thanks for the advice, add a new case in IndexShuffleBlockResolverSuite.scala. Done in 3ec15dc.